cibuildwheel/options.py

```diff
@@ -339,9 +339,11 @@ def config_file_path(self) -> Optional[Path]:
             return Path(args.config_file.format(package=args.package_dir))
 
         # return pyproject.toml, if it's available
-        pyproject_toml_path = Path(args.package_dir) / "pyproject.toml"
-        if pyproject_toml_path.exists():
-            return pyproject_toml_path
+        package_dir = Path(args.package_dir)
```

Comment:
Can't we pull this out of the tar.gz? Or even zip, but I'm fine in following build and only allowing .tar.gz's.

(There's also the temporary dir idea - the only benefit of this I can immediately think of is just validation, but if the build tools support it (and build should at some point), this seems cleaner)

Reply:
This is extracted now.